lib/common/patchset.c

```diff
@@ -476,11 +476,12 @@ xml_log_patchset_v1(uint8_t log_level, const xmlNode *patchset)
         options |= xml_log_option_diff_short;
     }
 
+    // @TODO: Does "- " or "+ " ever *not* get overridden by "--" or "++"?
```

Comment:
I haven't tested for this yet (and it seems like it would require a little bit of hacking since this is for v1 only). The logging paradigm here is just odd-looking and I'm not sure it's ideal. If I read it right...

We start with prefix `"- "` (or `"+ "`). Then if the element either (a) has no children or (b) possesses the `XML_DIFF_MARKER` attribute, we set `xml_log_option_diff_all`, change the prefix to `"--"` (or `"++"`), and log the whole subtree via `pcmk__xml_log().`

Otherwise:
* If `xml_log_option_diff_short` is set (meaning `log_level < LOG_DEBUG`), we recurse through the children and repeat the process.
* Else, we log the whole subtree with the original prefix `"- "` (or `"+ "`).

This PR aims to keep the existing behavior for now.

Reply:
In practice, no one is doing a rolling upgrade from 1.1.11 to 2.1.x so I wouldn't sweat it